pkg/cloud/services/ec2/instances.go

```diff
@@ -182,20 +182,7 @@ func (s *Service) CreateInstance(scope *scope.MachineScope, userData []byte, use
 	}
 	input.SubnetID = subnetID
 
-	if ptr.Deref(scope.AWSMachine.Spec.PublicIP, false) {
-		subnets, err := s.getFilteredSubnets(&ec2.Filter{
-			Name:   aws.String("subnet-id"),
-			Values: aws.StringSlice([]string{subnetID}),
-		})
-		if err != nil {
-			return nil, fmt.Errorf("could not query if subnet has MapPublicIpOnLaunch set: %w", err)
-		}
-		if len(subnets) == 0 {
-			return nil, fmt.Errorf("expected to find subnet %q", subnetID)
-		}
-		// If the subnet does not assign public IPs, set that option in the instance's network interface
-		input.PublicIPOnLaunch = ptr.To(!aws.BoolValue(subnets[0].MapPublicIpOnLaunch))
-	}
+	input.PublicIPOnLaunch = scope.AWSMachine.Spec.PublicIP
```

Comment:
@r4f4 the PR 4905 should be the draft to propose the BYOIP changes.

I have one question for this line/change:
Considering:
- subnet map on launch is added by default on CAPA deployments
- user's config for PublicIP on MachineSpec would have precedence, and may not affect, or depend, if subnet has set the field or not

Is there any specific use case to discover the subnets's map on launch attribute to take the decision of setting the MapPublicIpOnLaunch on instance create, instead of always setting it?

Reply:
I don't think so.